tests/execution/test-union-fieldtype.md

```diff
@@ -0,0 +1,192 @@
+# Test union type resolve
+
+```graphql @config
+schema @server @upstream(baseURL: "http://jsonplaceholder.typicode.com") {
+  query: Query
+}
+
+union FooBar = Bar | Foo
+
+type Bar {
+  bar: String!
+}
+
+type Foo {
+  foo: String!
+}
+
+type Nested {
+  bar: FooBar @discriminate(field: "type")
+  foo: FooBar @discriminate(field: "type")
+}
+
+type Query {
+  arr: [FooBar] @http(path: "/arr") @discriminate(field: "type")
+  bar: FooBar @http(path: "/bar") @discriminate(field: "type")
+  foo: FooBar @http(path: "/foo") @discriminate(field: "type")
```

Comment:
is there any validation that this field is present for every type that is part of the union?

Reply:
The `field` is **not required** to be part of the GraphQL schema, so I cannot check if it is a member of every union. However, the TypeFieldDiscriminator validates that this field is present in the JSON response and gives the appropriate error message if it is missing.